Cython/Utility/ExtensionTypes.c

```diff
@@ -205,6 +215,12 @@ static int __Pyx_PyType_Ready(PyTypeObject *t) {
 
 #else
     int r;
+
+    if (!__Pyx_PyType_HasMultipleInheritance(t)) {
+        // shortcut - if none of the base classes do multiple inheritance then we don't need to
+        // (and shouldn't) mess around with faking heaptypes.
+        return PyType_Ready(t);
```

Comment:
Don't we still need to set `Py_TPFLAGS_IMMUTABLETYPE`?

Reply:
No I don't think so. That's automatically set for static types. It only ends up unset in the hacky path because we lie to Python and tell it that it's a heaptype.